`/paper dumpitem` in Terminal Console case is not catched

<!-- Before reporting an issue here, please make sure you are on the latest version of Paper. -->
<!-- Forks of Paper receive no support here. Please make sure that your issue is reproducable on Paper first.  -->

Running the command `/paper dumpitem` causes a stack trace.

### What behaviour is expected:
The message `Only players can use this command`

### What behaviour is observed:
The stacktrace:
```
[XX:XX:XX WARN]: Unexpected exception while parsing console command "paper dumpitem"
org.bukkit.command.CommandException: Unhandled exception executing 'paper dumpitem' in com.destroystokyo.paper.PaperCommand(paper)
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:169) ~[patched_1.16.4.jar:git-Paper-393]
        at org.bukkit.craftbukkit.v1_16_R3.CraftServer.dispatchCommand(CraftServer.java:807) ~[patched_1.16.4.jar:git-Paper-393]
        at org.bukkit.craftbukkit.v1_16_R3.CraftServer.dispatchServerCommand(CraftServer.java:769) ~[patched_1.16.4.jar:git-Paper-393]
        at net.minecraft.server.v1_16_R3.DedicatedServer.handleCommandQueue(DedicatedServer.java:411) ~[patched_1.16.4.jar:git-Paper-393]
        at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:378) ~[patched_1.16.4.jar:git-Paper-393]
        at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1204) ~[patched_1.16.4.jar:git-Paper-393]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:992) ~[patched_1.16.4.jar:git-Paper-393]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:173) ~[patched_1.16.4.jar:git-Paper-393]
        at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: java.lang.ClassCastException: class com.destroystokyo.paper.console.TerminalConsoleCommandSender cannot be cast to class org.bukkit.craftbukkit.v1_16_R3.entity.CraftPlayer (com.destroystokyo.paper.console.TerminalConsoleCommandSender and org.bukkit.craftbukkit.v1_16_R3.entity.CraftPlayer are in unnamed module of loader 'app')
        at com.destroystokyo.paper.PaperCommand.doDumpItem(PaperCommand.java:173) ~[patched_1.16.4.jar:git-Paper-393]
        at com.destroystokyo.paper.PaperCommand.execute(PaperCommand.java:139) ~[patched_1.16.4.jar:git-Paper-393]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159) ~[patched_1.16.4.jar:git-Paper-393]
        ... 8 more
```

### Steps/models to reproduce:
1. Open the Paper Server Console
2. Type `paper dumpitem`
3. Hit Enter
4. ...?
5. Profit!

### Plugin list:
1. Dynmap (3.1-beta6-spigot)
2. AntiEndermanGrief (2020-07-21)

### Paper version:
```
[XX:XX:XX INFO]: This server is running Paper version git-Paper-393 (MC: 1.16.4) (Implementing API version 1.16.4-R0.1-SNAPSHOT)
[XX:XX:XX INFO]: Checking version, please wait...
[XX:XX:XX INFO]: Previous version: git-Paper-367 (MC: 1.16.4)
[XX:XX:XX INFO]: You are running the latest version
```

### Anything else:
As I am currently not able to modify the .patch file myself, here are the required lines in [Spigot-Server-Patches/0511-Paper-dumpitem-command.patch Line 53](https://github.com/PaperMC/Paper/blob/master/Spigot-Server-Patches/0511-Paper-dumpitem-command.patch#L53)
```patch
+        if (!(sender instanceof Player)) {
+            sender.sendMessage("Only players can use this command");
+            return;
+        }
```
It might be handy to move this into a dedicated function as `private void doFixLight(CommandSender sender, String[] args)` uses the same lines as the command `/paper fixlight` can also only be run by players.

--- a/Spigot-Server-Patches/0511-Paper-dumpitem-command.patch
+++ b/Spigot-Server-Patches/0511-Paper-dumpitem-command.patch
@@ -6,7 +6,7 @@ Subject: [PATCH] Paper dumpitem command
 Let's you quickly view the item in your hands NBT data
 
 diff --git a/src/main/java/com/destroystokyo/paper/PaperCommand.java b/src/main/java/com/destroystokyo/paper/PaperCommand.java
-index f60800c3bc06a38493e17b00b815f18cb87cc8bf..196648e65c6f58c18e19623474d69d780d542689 100644
+index f60800c3bc06a38493e17b00b815f18cb87cc8bf..0abfe19e204d20af0f8dedbeedb0ef98dfe9d3c8 100644
 --- a/src/main/java/com/destroystokyo/paper/PaperCommand.java
 +++ b/src/main/java/com/destroystokyo/paper/PaperCommand.java
 @@ -21,7 +21,9 @@ import org.bukkit.command.CommandSender;
@@ -46,11 +46,15 @@ index f60800c3bc06a38493e17b00b815f18cb87cc8bf..196648e65c6f58c18e19623474d69d78
              case "debug":
                  doDebug(sender, args);
                  break;
-@@ -164,6 +169,19 @@ public class PaperCommand extends Command {
+@@ -164,6 +169,23 @@ public class PaperCommand extends Command {
          return true;
      }
  
 +    private void doDumpItem(CommandSender sender) {
++        if (!(sender instanceof Player)) {
++            sender.sendMessage("Only players can use this command");
++            return;
++        }
 +        ItemStack itemInHand = ((CraftPlayer) sender).getItemInHand();
 +        net.minecraft.server.ItemStack itemStack = CraftItemStack.asNMSCopy(itemInHand);
 +        NBTTagCompound tag = itemStack.getTag();

